gtda/utils/tests/test_validation.py

```diff
@@ -54,3 +56,206 @@ def test_inputs_arrayStruc_V():
+    should be thrown by check_point_clouds."""
+
+    ex = CreateInputs(
+        n_samples, n_1, n_2, n_samples_extra, n_1_extra, n_2_extra).\
+        insert_inf()
+
+    check_point_clouds(ex.X, distance_matrices=True)
+    check_point_clouds(ex.X_list, distance_matrices=True)
+    check_point_clouds(ex.X_rectang, force_all_finite=False)
+    check_point_clouds(ex.X_list_rectang, force_all_finite=False)
+
+
+def test_check_point_clouds_value_err_inf():
+    """Cases in which part of the input is infinite and we throw a
+    ValueError."""
+
+    ex = CreateInputs(
+        n_samples, n_1, n_2, n_samples_extra, n_1_extra, n_2_extra).\
+        insert_inf()
+
+    # Check that, by default, np.inf is only accepted when distance_matrices
+    # is True.
+    # 1) Array input
+    with pytest.raises(ValueError):
+        check_point_clouds(ex.X_rectang)
+    # 2) List input
+    with pytest.raises(ValueError):
+        check_point_clouds(ex.X_list_rectang)
+
+    # Check that we error if we explicitly set force_all_finite to True
+    # 1) Array input
+    with pytest.raises(ValueError):
+        check_point_clouds(ex.X, distance_matrices=True, force_all_finite=True)
+    # 2) List input
+    with pytest.raises(ValueError):
+        check_point_clouds(
+            ex.X_list, distance_matrices=True, force_all_finite=True)
+
+
+def test_check_point_clouds_regular_nan():
+    """Cases in which part of the input is NaN and no warnings or errors
+    should be thrown by check_point_clouds."""
+
+    ex = CreateInputs(
+        n_samples, n_1, n_2, n_samples_extra, n_1_extra, n_2_extra).\
+        insert_nan()
+
+    check_point_clouds(ex.X, distance_matrices=True,
+                       force_all_finite='allow-nan')
+    check_point_clouds(
+        ex.X_list, distance_matrices=True, force_all_finite='allow-nan')
+    check_point_clouds(ex.X_rectang, force_all_finite='allow-nan')
+    check_point_clouds(ex.X_list_rectang, force_all_finite='allow-nan')
+
+
+@pytest.mark.parametrize("force_all_finite", [True, False])
+def test_check_point_clouds_value_err_nan(force_all_finite):
+    """Cases in which part of the input is nan and we throw a
+    ValueError."""
+
+    ex = CreateInputs(
+        n_samples, n_1, n_2, n_samples_extra, n_1_extra, n_2_extra).\
+        insert_nan()
+
+    # Check that we error when force_all_finite is True or False
+    # 1) Array input
+    with pytest.raises(ValueError):
+        check_point_clouds(
+            ex.X, distance_matrices=True, force_all_finite=force_all_finite)
+    with pytest.raises(ValueError):
+        check_point_clouds(ex.X_rectang, force_all_finite=force_all_finite)
+    # 2) List input
+    with pytest.raises(ValueError):
+        check_point_clouds(ex.X_list, distance_matrices=True,
+                           force_all_finite=force_all_finite)
+    with pytest.raises(ValueError):
+        check_point_clouds(
+            ex.X_list_rectang, force_all_finite=force_all_finite)
```

Comment:
I don't understand why these pass: why will `check_point_clouds` throw an error, if `force_all_finite=False`?

Reply:
Correct, because there are NaNs this time. The behaviour is different from sklearn's `check_array`, and is the reason for creating `_check_array_mod`. In `check_array`, `force_all_finite=False` means that both `np.nan` and `np.inf` is allowed, while `force_all_finite='allow-nan'` means that `np.nan` is allowed but `np.inf` is not. I thought a more useful behaviour for us when `force_all_finite=False` would be: `np.inf` is OK, but `np.nan` is not.

Reply:
Thank you for the detailed explanation, it's clear.